packages/flame/lib/src/components/component_notifier.dart

```diff
@@ -0,0 +1,39 @@
+import 'package:flame/components.dart';
+import 'package:flutter/material.dart';
+import 'package:meta/meta.dart';
+
+/// A [ChangeNotifier] that notifies its listeners when a [Component] is
+/// added or removed, or updated.
+class ComponentNotifier<T extends Component> extends ChangeNotifier {
+  ComponentNotifier(List<T> initial) : _components = initial;
+
+  final List<T> _components;
+
+  /// The list of components.
+  List<T> get components => _components;
+
+  /// Returns a single element of the components on the game.
+  ///
+  /// Returns null if there is no component.
+  T? get single {
+    if (_components.isNotEmpty) {
+      return _components.first;
+    }
+    return null;
+  }
+
+  @internal
+  void add(T component) {
+    _components.add(component);
+    notifyListeners();
```

Comment:
Is there a way to make the notification also carry some information about what has happened, which component added, which removed, or whether it was an "update" (which I guess may too carry some info).

Reply:
Not really since this is `ChangeNotifier`, its API only notifies changes, getting what changes is the responsibility of the user of the notifier instance.